src/components/ExportCapture.js

```diff
@@ -0,0 +1,122 @@
+    margin: theme.spacing(2, 1),
+    flexGrow: 1,
+  },
+  formControl: {
+    margin: theme.spacing(3),
+  },
+  csvLink: {
+    textDecoration: "none",
+    color: theme.palette.stats.green,
+  }
+}));
+
+const ExportCapture = (props) => {
+  const { isOpen, handleClose, columns, filter, speciesState } = props;
+  const classes = useStyle();
+  let nameColumns = {};
+  columns.forEach(({ attr }) => {
+    nameColumns[attr] = true;
+  });
+  const [checkedColumns, setCheckColumns] = useState(nameColumns);
+  const capturesContext = useContext(CapturesContext);
+  const [downloadData, setDownloadData] = useState([]);
+
+  function handleChange(attr) {
+    setCheckColumns({ ...checkedColumns, [attr]: false });
+  }
+
+  function processDownloadData(captures, selectedColumns) {
+    return captures.map(capture => {
+      let formatCapture = {};
+      Object.keys(selectedColumns).forEach(attr => {
+        if (attr === 'id' || attr === 'planterId') {
+          formatCapture[attr] = capture[attr];
+        } else if (attr === 'tokenId') {
+          formatCapture[attr] = formatCell(capture, speciesState, attr, (val) =>
+          val ? tokenizationStates.TOKENIZED : tokenizationStates.NOT_TOKENIZED);
+        } else if (attr === 'timeCreated') {
+          formatCapture[attr] = formatCell(capture, speciesState, attr, (val) => getDateTimeStringLocale(val));
+        } else {
+          formatCapture[attr] = formatCell(capture, speciesState, attr, null);
+        }
+      })
+      return formatCapture;
+    })
+  }
+
+  return (
+    <Dialog
+      open={isOpen}
+      onClose={handleClose}
+      aria-labelledby="form-dialog-title"
+    >
+      <DialogTitle id="form-dialog-title">Export Captures</DialogTitle>
+      <DialogContent>
+        <FormControl component="fieldset" className={classes.formControl}>
+          <FormGroup>
+            {columns.map(({ attr, label }) => (
+              <FormControlLabel
+                key={attr}
+                control={
+                  <Checkbox
+                    checked={checkedColumns[attr]}
+                    onChange={() => handleChange(attr)}
+                    name={attr}
+                  />
+                }
+                label={label}
+              />
+            ))}
+          </FormGroup>
+        </FormControl>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={handleClose}>Cancel</Button>
+        <Button color="primary">
+          <CSVLink
+            className={classes.csvLink}
+            data={downloadData}
+            filename={"captures.csv"}
+            asyncOnClick={true}
+            onClick={() => {
+              let filterColumns = Object.entries(checkedColumns).filter((val) => val[1] === true);
+              const selectedColumns = Object.fromEntries(filterColumns);
+              capturesContext.getAllCaptures({ filter }).then((response) => {
+                setDownloadData(processDownloadData(response.data, selectedColumns));
```

Comment:
`processDownloadData` takes a long time and locks up the browser for a few seconds on a fairly powerful machine. On a slower machine this could be a lot longer.

As a minimum, we need to give the user visual feedback that the export is in progress, but even better would be to do the processing in a web worker.
If it's helpful, we can ask the Design team to create a visual workflow.